### ISSUE TITLE
Create custom metrics using R functions

### ISSUE THREAD

**Issue by mattwarkentin** [rstudio/keras#1122]
Hi,

I am interested in making a custom metric to compute AUC at the end of each batch (rather than computing accuracy). I have already figured out how to set up a callback using `callback_lambda()` to compute AUC `on_epoch_end`...

```r
# This works as a callback
auc <- vector("numeric")
callback_auc <- callback_lambda(
  on_epoch_end = function(batch, logs) {
    epoch_auc <- yardstick::roc_auc_vec(yy, predict(model, xx))
    auc <<- c(auc, epoch_auc)
  }
)
```

But I was hoping to compute AUC as a custom metric instead by supplying it to `compile(..., metric = auc)`. It seems like something like this should work for creating the custom metric:
```r
auc <- custom_metric("auc", function(y_true, y_pred) {yardstick::roc_auc_vec(y_true, y_pred)})

model %>% compile(
  optimizer = 'rmsprop',
  loss = 'categorical_crossentropy',
  metric = auc
)
```

But it fails. Besides the fact that `{yardstick}` requires `y_true` to be a factor, I believe the salient issue is that `y_true` and `y_pred` are returned as `tensorflow` tensors and not `R` vectors, and I am really not sure how to handle these tensors when trying to use `R` functions. Any guidance on how to create a custom metric which use R functions internally?

**Comment by AshesITR** [comment]
You can convert tensors to R objects via `y_true$numpy()`, `as.numeric(y_true)`, `as.matrix(y_true)` or `as.array(y_true)` depending on your usecase and the rank of `y_true`.
